### DISCUSSION TITLE
Understanding Gusto

### DISCUSSION THREAD

**Discussion by adominious** [firedrakeproject/gusto#422] [General]
Hi everyone. I am very new to the climate scene and I was hoping for some advice about where is best to look/research in order to best understand the core components of this package and dry dynamical cores as a whole.

I have experience working with PDEs and the FEM, but my Continuum Mechanics is quite poor. Any help/advice would be most appreciated. Thank you!

**Comment by jshipton** [comment]
Hi @adominious! We're in the process of writing a paper about Gusto but I'm afraid it is not yet ready to share. [This paper](https://www.cambridge.org/core/journals/acta-numerica/article/compatible-finite-element-methods-for-geophysical-fluid-dynamics/D2A87A43005D82261B9C7C4BB7CD000E) is an excellent review of compatible finite element methods for geophysical fluid dynamics and covers the methods that Gusto implements (and more!). Hope that helps get you started...

**Reply by adominious** [reply]
Hi again.

Thank you for this information. I am specifically looking to work with the Primitive Equations. I don't suppose there is a particular file I can run that will give me an output of running these, given some initial conditions?

**Reply by jshipton** [reply]
Hi @adominious! We haven't got an off-the-shelf solver for the primitive equations I'm afraid. Not that it'd be too tricky, but it's not on our immediate to do list I'm afraid. What kind of questions are you trying to answer? We can provide compatible finite element disrectisations of the shallow water equations, the incompressible Boussinesq equations and the compressible Euler equations - are any of those of use to you for the situation that you are modelling?

**Comment by adominious** [comment]
I wanted to compare the outputs of a DDC (modeled by the primitive equations) vs a full wet NWP model (I should note I am very much so a novice with all the terminology, so I apologise if some of this does not make much sense).

**Reply by jshipton** [reply]
What's a DDC? I don't think we can call Gusto a full NWP model I'm afraid! It might be better for us to have a chat with you about what it is you want to do - it sounds like Gusto might not be the right thing for it at this stage.

**Reply by adominious** [reply]
A Dry Dynamical Core, an atmospheric model that strips away all the sub-grid moisture processes, and is modelled by the Primitive Equations. I am aware gusto is only a dynamical core model, I am using a different model for the NWP part :)

Do you think Gusto is suitable for solving the Primitive Equations? Could something like this be ‘built’?
